metals/src/main/scala/scala/meta/internal/metals/logging/MetalsLogger.scala

```diff
@@ -113,12 +116,35 @@ object MetalsLogger {
       redirectSystemStreams: Boolean,
   ): Unit = {
     val newLogFiles = folders.map(_.resolve(workspaceLogPath))
+    newLogFiles.foreach(truncateLogFile)
     scribe.info(s"logging to files ${newLogFiles.mkString(",")}")
     if (redirectSystemStreams) {
       redirectSystemOut(newLogFiles)
     }
   }
 
+  private def truncateLogFile(path: AbsolutePath) = {
+    val MaxLines = 10000
+    if (Files.exists(path.toNIO)) {
+      try {
+        def linesWithIndices =
+          Files.newBufferedReader(path.toNIO).lines().asScala.zipWithIndex
```

Comment:
This will get all the lines into the memory I think no? Which might be problematic in case somebody has a very long file already :thinking: 

What if instead we kept the last 10 sessions? Just backup the last file with timestamp and then remove the oldest.

The other option would be using `tail -n 10000`, but not sure if that would work for Windows (most likely not :/)

Reply:
`lines().asScala.zipWithIndex` returns a generator, so I was hoping it won’t try to read all the lines into the memory if they don’t fit. We keep the last 10k in the memory.